inc/database/processors/tmsconnect/class-tmsconnect-media-processor.php

```diff
@@ -0,0 +1,66 @@
+<?php
+/**
+ * The class used to process TMS Media Attachments
+ */
+namespace TMSC\Database\Processors\TMSConnect;
+class TMSConnect_Media_Processor extends \TMSC\Database\TMSC_Processor {
+	/**
+	 * Which migratable type the objects of this processor will be.
+	 */
+	public $migrateable_type = 'Media';
+
+	/**
+	 * The key used for the current attachment query
+	 * @var string
+	 */
+	public $object_query_key = 'tms_attachments';
+
+	/**
+	 * The parent object for this set of media attachments.
+	 */
+	public $parent_object;
+
+	/**
+	 * Current object raw data.
+	 * @var object
+	 */
+	private $current_object = null;
+
+	/**
+	 * Constructor
+	 * @param string $type
+	 */
+	public function __construct( $type ) {
+		parent::__construct( $type );
+		add_filter( 'tmsc_get_child_processors', array( $this, 'set_child_processors' ) );
+	}
+
+	/**
+	 * This is a child processor so register it as such.
+	 * @param array $child_processors.
+	 * @return array.
+	 */
+	public function set_child_processors( $child_processors ) {
+		$child_processors[] = $this->processor_type;
+		return $child_processors;
+	}
+
+	/**
+	 * Run our import in batches by taxonomy.
+	 * @return void
+	 */
+	public function run() {
+		parent::run();
```

Comment:
I dont think you need to have this function here since it just calls the parent function.  If you remove it, the parent function will be called anyways and you will have the same functionality.